cai_causal_graph/time_series_causal_graph.py

```diff
@@ -0,0 +1,883 @@
+        destination: Optional[NodeLike] = None,
+        *,
+        edge_type: EDGE_T = EDGE_T.DIRECTED_EDGE,
+        meta: Optional[dict] = None,
+        edge: Optional[Edge] = None,
+        **kwargs,
+    ) -> Edge:
+        """
+        Add an edge to the graph. See `cai_causal_graph.causal_graph.CausalGraph.add_edge` for more details.
+
+        In addition to the `CausalGraph.add_edge` method, this method also populates the metadata of the nodes with
+        the variable name and the time lag.
+        """
+        # if the source and destination time series nodes do not exist, create them
+        if source is not None and not self.node_exists(source):
+            source_meta = None
+            if isinstance(source, HasMetadata):
+                source_meta = source.get_metadata()
+                source = self.add_node(source, meta=source_meta)
+
+        if destination is not None and not self.node_exists(destination):
+            destination_meta = None
+            if isinstance(destination, HasMetadata):
+                destination_meta = destination.get_metadata()
+                destination = self.add_node(destination, meta=destination_meta)
+
+        edge = super().add_edge(source, destination, edge_type=edge_type, meta=meta, edge=edge, **kwargs)
+
+        return edge
+
+    @classmethod
+    def from_causal_graph(cls, causal_graph: CausalGraph) -> TimeSeriesCausalGraph:
+        """
+        Return a time series causal graph from a causal graph.
+
+        This is useful for converting a causal graph from a single time step into a time series causal graph.
+
+        :param causal_graph: The causal graph.
+        """
+
+        sepsets = deepcopy(causal_graph._sepsets)
+
+        # copy nodes and make them TimeSeriesNodes
+        ts_cg = cls()
+        for node in causal_graph.get_nodes():
+            meta = deepcopy(node.meta)
+            # get the variable name and lag from the node name
+            variable_name, lag = get_variable_name_and_lag(node.identifier)
+            node = TimeSeriesNode(variable_name=variable_name, time_lag=lag, meta=meta)
+            ts_cg.add_node(node)
+
+        # copy edges
+        for edge in causal_graph.get_edges():
+            source = ts_cg.get_node(edge.source)
+            destination = ts_cg.get_node(edge.destination)
+            ts_cg.add_edge(source, destination, meta=edge.meta)
+
+        ts_cg._sepsets = sepsets
+
+        return ts_cg
+
+    @staticmethod
+    def from_adjacency_matrix(
+        adjacency_matrix: numpy.ndarray,
+        node_names: Optional[List[Union[NodeLike, int]]] = None,
+    ) -> TimeSeriesCausalGraph:
+        """
+        Return a time series causal graph from an adjacency matrix.
+        This is useful for converting an adjacency matrix into a time series causal graph.
+        """
+        graph = CausalGraph.from_adjacency_matrix(adjacency_matrix, node_names=node_names)
+        return TimeSeriesCausalGraph.from_causal_graph(graph)
```

Comment:
Will this actually check that time order is respected? Need to test this.

Reply:
The TimeSeriesCausalGraph should do these checks, there aren't any checks I believe as I had a comment somewhere else on this issue

Reply:
I added fix to `add_edge` and improved unit tests to check on adding edge and on constructing from causal graph object.